src/components/Microsoft.DotNet.UpgradeAssistant.MSBuild/MSBuildTargetTFMSelector.cs

```diff
@@ -68,36 +69,53 @@ public TargetFrameworkMoniker SelectTFM(IProject project)
             var tfm = new TargetFrameworkMoniker(tfmName);
             foreach (var dep in project.ProjectReferences)
             {
-                try
+                foreach (var depTFM in dep.TargetFrameworks)
                 {
-                    if (_tfmComparer.IsCompatible(tfm, dep.TFM))
+                    try
                     {
-                        continue;
+                        if (_tfmComparer.IsCompatible(tfm, depTFM))
+                        {
+                            continue;
```

Comment:
This isn't quite right. If the dependency has any TFMs that are compatible with our proposed TFM we should continue out of the *outer* loop (not just the inner one) since that's sufficient for the proposed TFM to make sense with the dependency.

Reply:
Rewrote and added tests